simulator/watcher.go

```diff
@@ -24,16 +24,15 @@ func newWatcher(ch chan *object.Object) (watcher, error) {
 
 func (w *watcher) sendIfNeeded(obj *object.Object) error {
 	for _, k := range w.keys {
-		if !k.Match(obj.Key) {
-			continue
+		if k.Match(obj.Key) {
```

Comment:
This approach causes more indentations. It's better to check exceptions, firstly, and indent lines of code as few as possible

Reply:
So you are sure that in the most cases keys are not matched.